pkgs/development/python-modules/datalad-next/default.nix

```diff
@@ -0,0 +1,106 @@
+{
+  annexremote,
+  buildPythonPackage,
+  datalad,
+  datasalad,
+  fetchFromGitHub,
+  git,
+  git-annex,
+  humanize,
+  lib,
+  more-itertools,
+  psutil,
+  pythonOlder,
+  setuptools,
+  openssh,
+  unzip,
+  webdavclient3,
+}:
+
+buildPythonPackage rec {
+  pname = "datalad-next";
+  version = "1.5.0";
+  pyproject = true;
+  disabled = pythonOlder "3.12";
+
+  src = fetchFromGitHub {
+    owner = "datalad";
+    repo = "datalad-next";
+    rev = "refs/tags/${version}";
+    hash = "sha256-fqP6nG2ncDRg48kvlsmPjNBOzfQp9+7wTcGvsYVrRzA=";
+  };
+
+  nativeBuildInputs = [ git ];
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    annexremote
+    datasalad
+    datalad
+    humanize
+    more-itertools
+  ];
+
+  preCheck = ''
+    export HOME=$TMPDIR
+  '';
+
+  nativeCheckInputs = [
```

Comment:
Yes, but the dosen't work this way. You need to add `pytestCheckHook`. I've run your code locally and it doesn't run tests. You can run it with and without it and see the difference.

https://nixos.org/manual/nixpkgs/unstable/#contributing-guidelines
> Make sure the tests are enabled using for example [pytestCheckHook](https://nixos.org/manual/nixpkgs/unstable/#using-pytestcheckhook) and, in the case of libraries, are passing for all interpreters. If certain tests fail they can be disabled individually. Try to avoid disabling the tests altogether. In any case, when you disable tests, leave a comment explaining why.

Reply:
Phases without hook:

```
...
Running phase: unpackPhase
...
Running phase: patchPhase
...
Running phase: updateAutotoolsGnuConfigScriptsPhase
...
Running phase: configurePhase
...
Running phase: buildPhase
...
Running phase: pythonRuntimeDepsCheckHook
...
Running phase: installPhase
...
Running phase: pythonOutputDistPhase
...
Running phase: fixupPhase
...
Running phase: installCheckPhase
...
Running phase: pythonCatchConflictsPhase
...
Running phase: pythonRemoveBinBytecodePhase
...
Running phase: pythonImportsCheckPhase
...
```

Phases with hook:

```
...
Running phase: unpackPhase
...
Running phase: patchPhase
...
Running phase: updateAutotoolsGnuConfigScriptsPhase
...
Running phase: configurePhase
...
Running phase: buildPhase
...
Running phase: pythonRuntimeDepsCheckHook
...
Running phase: installPhase
...
Running phase: pythonOutputDistPhase
...
Running phase: fixupPhase
...
Running phase: installCheckPhase
...
Running phase: pythonCatchConflictsPhase
...
Running phase: pythonRemoveBinBytecodePhase
...
Running phase: pythonImportsCheckPhase
...
Running phase: pytestCheckPhase
Executing pytestCheckPhase
evaling implicit 'preCheck' string hook
============================= test session starts ==============================
platform linux -- Python 3.12.4, pytest-8.2.2, pluggy-1.5.0
rootdir: /build/source
configfile: pyproject.toml
collected 451 items / 32 deselected / 419 selected                             
datalad_next/annexbackends/tests/test_base.py ..                         [  0%]
datalad_next/annexremotes/tests/test_archivist.py ..                     [  0%]
...
Running phase: pytestcachePhase
Running phase: pytestRemoveBytecodePhase
```

Reply:
It seems you are right. I'm a bit confused, because I disabled some tests because of the executed pytest phase. When I now run it again, it actually does not run the tests..Who knows. 
I'll just add the `pytestCheckHook`